flang/lib/Lower/Bridge.cpp

```diff
@@ -1020,7 +1045,7 @@ class FirConverter : public Fortran::lower::AbstractConverter {
         const auto *hostDetails =
             sym->detailsIf<Fortran::semantics::HostAssocDetails>();
         assert(hostDetails && "missing local_init variable host variable");
-        [[maybe_unused]] const Fortran::semantics::Symbol &hostSym =
+        LLVM_ATTRIBUTE_UNUSED const Fortran::semantics::Symbol &hostSym =
```

Comment:
What is the rational for changing `[[maybe_unused]]` to `LLVM_ATTRIBUTE_UNUSED` ?

`[[maybe_unused]]` is standard C++ 17, `LLVM_ATTRIBUTE_UNUSED` is not standard (It uses `__attribute__((__unused__))`, see https://github.com/llvm/llvm-project/blob/a1862d78eb4502405fa2c25b100168b66c6d071a/llvm/include/llvm/Support/Compiler.h#L180) and is actually advised against for objects (comment above the macro definition). I understand the LLVM code base that is not C++17 cannot use `[[maybe_unused]]`, but since flang code base is C++ 17, I would have imagined we should favor the standard way of doing it.

Reply:
Yes, I understand.

We had a mix of both and since this code is following LLVM coding standards, it made sense to follow that coding standard and defer to their macros, data structures, etc. over similar, new-ish C++ syntax.